tests/testTriggers.go

```diff
@@ -7319,6 +7319,339 @@ func TriggerMetroDR(contexts *[]*scheduler.Context, recordChan *chan *EventRecor
 	updateMetrics(*event)
 }
 
+// TriggerAsyncDRPXRestartSource triggers Async DR with PX restart on source
+func TriggerAsyncDRPXRestartSource(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer endLongevityTest()
+	startLongevityTest(AsyncDRPXRestartSource)
+	defer ginkgo.GinkgoRecover()
+	log.Infof("Async DR PX restart on source trigger triggered at: %v", time.Now())
+	defer ginkgo.GinkgoRecover()
+	event := &EventRecord{
+			Event: Event{
+					ID:   GenerateUUID(),
+					Type: AsyncDRPXRestartSource,
+			},
+			Start:   time.Now().Format(time.RFC1123),
+			Outcome: []error{},
+	}
+	defer func() {
+			event.End = time.Now().Format(time.RFC1123)
+			*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	chaosLevel := ChaosMap[AsyncDRPXRestartSource]
+	var (
+			migrationNamespaces   []string
+			taskNamePrefix        = "async-dr-pxrs"
+			allMigrations         []*storkapi.Migration
+			includeVolumesFlag    = true
+			includeResourcesFlag  = true
+			startApplicationsFlag = false
+	)
+
+	Step(fmt.Sprintf("Deploy applications for migration, with frequency: %v", chaosLevel), func() {
+
+			// Write kubeconfig files after reading from the config maps created by torpedo deploy script
+			err := asyncdr.WriteKubeconfigToFiles()
+			if err != nil {
+					log.Errorf("Failed to write kubeconfig: %v", err)
+			}
+
+			err = SetSourceKubeConfig()
+			if err != nil {
+					log.Errorf("Failed to Set source kubeconfig: %v", err)
+			}
+			UpdateOutcome(event, err)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+					taskName := fmt.Sprintf("%s-%d-%s", taskNamePrefix, i, time.Now().Format("15h03m05s"))
+					log.Infof("Task name %s\n", taskName)
+					appContexts := ScheduleApplications(taskName)
+					*contexts = append(*contexts, appContexts...)
+					ValidateApplications(*contexts)
+					for _, ctx := range appContexts {
+							// Override default App readiness time out of 5 mins with 10 mins
+							ctx.ReadinessTimeout = appReadinessTimeout
+							namespace := GetAppNamespace(ctx, taskName)
+							migrationNamespaces = append(migrationNamespaces, namespace)
+					}
+					Step("Create cluster pair between source and destination clusters", func() {
+							// Set cluster context to cluster where torpedo is running
+							ScheduleValidateClusterPair(appContexts[0], false, true, defaultClusterPairDir, false)
+					})
+			}
+
+			log.Infof("Migration Namespaces: %v", migrationNamespaces)
+
+	})
+
+	log.InfoD("Start migration")
+
+	for i, currMigNamespace := range migrationNamespaces {
+		migrationName := migrationKey + fmt.Sprintf("%d", i) + time.Now().Format("15h03m05s")
+		currMig, err := asyncdr.CreateMigration(migrationName, currMigNamespace, asyncdr.DefaultClusterPairName, currMigNamespace, &includeVolumesFlag, &includeResourcesFlag, &startApplicationsFlag)
+		if err != nil {
```

Comment:
you can use 
```
if err!=nil{
  UpdateOutcome
 continue
}
```

Reply:
I added a return key here, because we want failure even if first migration fails.

Reply:
seems return is missing

Reply:
I added.